episodes/itksnap.Rmd

```diff
@@ -0,0 +1,630 @@
+
+### NifTi format
+The Neuroimaging Informatics Technology Initiative (NIfTI) image format are usually stored as a single file containing the imaging data and header information.
+It also is possible to store the header and image data in separate files but this is not very common.
+
+There are actually two formats for nifti files, nifti-1 and nifti-2, with the later allowing for very large images. 
+However, the formats are very similar, and the header contains the same info (but some of it stored slightly differently). 
+They can be worked with and manipulated in the same way, and it is only when loading and saving the files (which is handled by NiBabel’s functions) that a distinction really needs to be made. 
+The nifti header contains much less information than DICOM, but does include all the key information required for interpreting, manipulating, and processing the image. 
+
+During these exercises you will learn about some of the key information stored in the nifti header, but for more information see: https://brainder.org/2012/09/23/the-nifti-file-format/ and https://brainder.org/2015/04/03/the-nifti-2-file-format/.
+
+## ITK-SNAP
+
+### Getting Started with ITK-SNAP
+#### History
+ITK-SNAP started in 1999 with SNAP (SNake Automatic Partitioning) and developed by Paul Yushkevich with the guidance of Guido Gerig. 
+ITK-SNAP is open-source software distributed under the GNU General Public License. 
+ITK-SNAP is written in C++ and it leverages the Insight Segmentation and Registration Toolkit (ITK) library. 
+
+#### Hardware Requirements
+* Memory: "Overall 2Gb of memory should be enough for most users; 4Gb should suffice for users with large images." [:link:](http://www.itksnap.org/pmwiki/pmwiki.php?n=Documentation.HardwareRequirements#:~:text=Memory%20usage%20in%20SNAP%20is,for%20a%20512%20cubed%20image.)
+* Multi - Operating System: 
+	* ITK-SNAP binaries are distributed for Linux (32 and 64 bit)), 
+	* Windows (64 bit), 
+	* and MacOS Leopard (64 bit).
+
+#### Installation
+##### Windows 11
+- Open the following link: https://sourceforge.net/projects/itk-snap/files/itk-snap/4.2.0/
+- Download `itksnap-4.2.0-20240422-win64-AMD64.exe` 
+- Install it by following default settings.
+
+##### Ubuntu 22.04 x64 GNU/Linux
+```
+FILENAME=itksnap-nightly-rel_4.0-Linux-gcc64.tar.gz  #Length: 200943059 (192M) [application/x-gzip]
+cd ~/Downloads/
+mkdir -p itksnap && cd itksnap
+wget https://sourceforge.net/projects/itk-snap/files/itk-snap/Nightly/$FILENAME/download -O $FILENAME
+tar -xvzf $FILENAME
+```
+
+### ITK-SNAP application
+ITK-SNAP application shows three orthogonal slices and a fourth window for three-dimensional view segmentation. 
+![ITK-SNAP main window](../fig/itk-snap-main-window.svg)
+
+
+## Viewing and understanding DICOM data
+
+We are using DICOM data from "3.000000-CT Lung  3.0  B31f-61322  [175 items, totalling 92.4 MB]".
+
+### Opening and Viewing DICOM images
+
+* Open DICOM images  
```

Comment:
step 3
we need to explain how to navigate around the image, i.e. change the displayed slices
explain this can be done by left clicking in the displayed slices, using the mouse wheel, or the slider next to the displayed slices
explain how you can zoom using the right mouse button (or control + button 1), and pan using the centre mouse button (or alt +  button 1)
(there may be other ways of doing the above as well - I have not used itksnap before so am still very much figuring it out myself!)
also need to explain that the cursor position (in voxels) and intensity value at the cursor are shown on the left in the Cursor Inspector
![image](https://github.com/HealthBioscienceIDEAS/Medical-Image-Registration-Short-Course/assets/9380572/432ccceb-1668-4c5a-85bb-1a583b538269)

when i wrote the comment above (now a few hours ago!) i was going to show how to adjust the intensity window here, but now I think that should be saved for when doing the colour overlay below

Reply:
step 4
images can also be opened by simply dragging them on to the itksnap window
in a file explorer (Finder in Mac) go to the exhale_BH_CT folder and drag any of the dicom files on to the itksnap window. A small window will pop up asking What should itksnap do with this image? If you select Load as Main Image it will replace the current image. Instead select Load as Additional Image - this will enable easy comparison of the two images.

Reply:
step 5 and 6
once the second image is loaded you will see two thumbnails in the top corner of each slice display
![image](https://github.com/HealthBioscienceIDEAS/Medical-Image-Registration-Short-Course/assets/9380572/b7a18f78-19f5-4d21-8d48-02ae0348da00)
click on these to swap the displayed slices from one image to the other - this will enable you to easily see the similarities and differences between the images
you will also see that both images now appear in the Cursor Inspector, where you can see the intensity value at the cursor for both images. you can also change the displayed image by clicking on it in the Cursor Inspector

Reply:
step 7
In some cases you may want to display an image as a colour overlay on top of another image, e.g. displaying a PET image overlaid on the corresponding CT.
Load the CT_for_PET image either using the drag and drop method or File->Open Image as described above
This time select Load as Main Image. This will load in the new image replacing the two that were previously loaded.

Reply:
step 8
To load the PET image as a colour overlay you need to use File->Add Another Image...
Click on Browse and select any one of the files in the PET folder. Make sure File Format is DICOM Image Series (it should be). Click Next and then Next again. When the image has loaded select As a semi-transparent overlay. you can also set the overlay color map here - select Hot from the dropdown menu and click Next. The image summary will then be displayed, along with a warning about possible loss of precision (which can be ignored). click Finish

Reply:
step 9 and 10
PET image appears in Cursor Inspector (called Galligas Lung) in italic. Right clicking on it opens a Menu that enables you to set the color map and change the opacity using a slider (amongst other things). 
Go to Tools->Layer Inspector. This opens the Image Layer Inspector window. you will see the two images on the left, and five tabs along the top of the window.
The General tab displays the filename and the nickname for the selected image. The Nickname can be edited. For the PET image the general tab also displays a slider to set the opacity, and the option to display it as a separate image or semi-transparent overlay
the Contrast tab enables you to adjust how the image intensities are displayed for the different images. e.g. select the lung image and set the maximum value to 1000 (push enter after typing the number) and this will make the structures in the lung easier to see.
The color map tab can be used to select and manipulate the colour map
The info tab displays the Image Header information for the images and also gives the Cursor Coordinates in both Voxel and World units. if you look at the info tab for both the CT image and PET image you will see that the header information for the images is different, and the cursor coordinates in voxel units for the PET image are not whole numbers - this is because the displayed slices are from the main (CT) image, and the overlaid (PET) image slices are interpolated at the corresponding locations.
finally the metadata tab contains (some of) the information from the DICOM header of the images

Reply:
FYI.  _... majority of jamie-mcclelland comments adding sections, figures and vector files with the combination of zsenousy instructions to avoid replication, also fixes datasets files_ [491c44d](https://github.com/HealthBioscienceIDEAS/Medical-Image-Registration-Short-Course/pull/5/commits/491c44d4eb7dad7e9e4e219e5a9d673f04ebe182)